src/Catel.Core/Catel.Core.Shared/Logging/LogListenerBase.cs

```diff
@@ -164,9 +165,9 @@ public TimeDisplay TimeDisplay
         /// <param name="time">The time.</param>
         void ILogListener.Write(ILog log, string message, LogEvent logEvent, object extraData, LogData logData, DateTime time)
         {
-            // We always want the Argument class logging, no matter what
-            var isArgumentLog = log.TargetType == ArgumentType;
-            if (!isArgumentLog)
+            // If the log is a catel log and the AlwaysLog flag is set, log the message.
+            var catelLog = log as ICatelLog;
+            if (!catelLog?.AlwaysLog ?? true)
```

Comment:
Not following; we don't enter the if statement if the AlwaysLog flag is true, if it's not true or not a CatelLog we do the additional checks before writing.  This is how it worked before, except instead of checking AlwaysLog we checked if it was coming from `Argument`.

Admittedly the logic is a little confusing as is, I can break it out.